raiden/transfer/channel.py

```diff
@@ -814,6 +831,114 @@ def is_valid_unlock(
     return result
 
 
+def is_valid_action_withdraw(
+    channel_state: NettingChannelState, withdraw: ActionChannelWithdraw
+) -> SuccessOrError:
+    result: SuccessOrError
+
+    balance = get_balance(sender=channel_state.our_state, receiver=channel_state.partner_state)
+
+    withdraw_amount = withdraw.total_withdraw - channel_state.our_state.total_withdraw
+    if withdraw_amount <= 0:
+        msg = "Total withdraw {} did not increase".format(withdraw.total_withdraw)
+        result = (False, msg)
+    elif balance < withdraw_amount:
+        msg = "Insufficient balance: {}. Requested {} for withdraw".format(
+            balance, withdraw_amount
+        )
+        result = (False, msg)
+    else:
+        result = (True, None)
+
+    return result
+
+
+def is_valid_withdraw_request(
+    channel_state: NettingChannelState, withdraw_request: ReceiveWithdrawRequest
+) -> SuccessOrError:
+    result: SuccessOrError
+
+    expected_nonce = get_next_nonce(channel_state.partner_state)
+    balance = get_balance(sender=channel_state.partner_state, receiver=channel_state.our_state)
+
+    packed = pack_withdraw(
+        canonical_identifier=withdraw_request.canonical_identifier,
+        participant=channel_state.partner_state.address,
+        total_withdraw=withdraw_request.total_withdraw,
+    )
+
+    valid_signature, signature_msg = is_valid_signature(
+        data=packed,
+        signature=withdraw_request.signature,
+        sender_address=channel_state.partner_state.address,
+    )
+
+    withdraw_amount = withdraw_request.total_withdraw - channel_state.partner_state.total_withdraw
+
+    if withdraw_amount <= 0:
+        msg = "Total withdraw {} did not increase".format(withdraw_request.total_withdraw)
+        result = (False, msg)
+    elif balance < withdraw_amount:
+        msg = "Insufficient balance: {}. Requested {} for withdraw".format(
+            balance, withdraw_amount
+        )
+        result = (False, msg)
+    elif not valid_signature:
+        result = (False, signature_msg)
+    elif withdraw_request.nonce != expected_nonce:
+        msg = (
+            f"Nonce did not change sequentially, expected: {expected_nonce} "
+            f"got: {withdraw_request.nonce}."
+        )
+
+        result = (False, msg)
+    else:
+        result = (True, None)
+
+    return result
+
+
+def is_valid_withdraw_confirmation(
```

Comment:
Please add the checks for the chain_id, token_address, channel_id, and for overflows.